contracts/cw20-dao-crowdfund/src/contract.rs

```diff
@@ -394,6 +403,48 @@ pub fn execute_receive_funding_tokens(
     }
 }
 
+pub fn get_fee_manager_config(
+    deps: &DepsMut,
+    state: &State,
+) -> Result<fee_manager::state::Config, StdError> {
+    let response: fee_manager::msg::ConfigResponse = deps.querier.query_wasm_smart(
+        state.fee_manager_addr.clone(),
+        &fee_manager::msg::QueryMsg::GetConfig {},
+    )?;
+    Ok(response.config)
+}
+
+// Ensure public payment funds were sent and create a message to forward them to the fee receiver.
+// Return a ContractError::InvalidPublicPayment if funds were not properly sent.
+pub fn take_public_payment(
+    deps: &DepsMut,
+    info: &MessageInfo,
+    state: &State,
+) -> Result<BankMsg, ContractError> {
+    // Get fee manager information.
+    let fee_manager_config = get_fee_manager_config(deps, state)?;
+
+    let payment = info
+        .funds
+        .iter()
+        .filter(|coin| coin.denom == fee_manager_config.public_listing_fee.denom)
+        .fold(Uint128::zero(), |accum, coin| coin.amount + accum);
+
+    if payment != fee_manager_config.public_listing_fee.amount {
+        return Err(ContractError::InvalidPublicPayment(format!(
+            "not equal to {}{}",
+            fee_manager_config.public_listing_fee.amount,
+            fee_manager_config.public_listing_fee.denom
+        )));
+    }
+
+    // Transfer public listing fee to the fee account.
+    Ok(BankMsg::Send {
+        to_address: fee_manager_config.receiver_addr.to_string(),
+        amount: vec![fee_manager_config.public_listing_fee.clone()],
+    })
```

Comment:
If the fee to create a campaign is zero this is going to return a bank message that transfers zero coins. This isn't an error persay, but it is likely a waste of gas fees. I think we should probably handle this case gracefully.